lucille-core/src/test/java/com/kmwllc/lucille/stage/DictionaryLookupTest.java

```diff
@@ -80,7 +83,139 @@ public void testGetLegalProperties() throws StageException {
             "dest",
             "conditions",
             "class",
-            "dict_path"),
+            "dict_path",
+            "set_only",
+            "ignore_missing_source"),
         stage.getLegalProperties());
   }
```

Comment:
I am pretty sure that we do not currently validate the stage configurations by default without running Lucille with a verification flag. I would like to revisit this decision and potentially verify stage configurations every time the stages are initialized. 

I am bringing this up because it took me a few minutes to find a mistake I made in the configuration of specifying the `setOnly` parameter instead of `set_only`.

Reply:
The `Stage` constructor line 88 calls validateConfigWithConditions() so we should be doing validation when any stage is constructed.  Jake, I think you were the one who implemented that :) Can you give steps for reproducing the problem you encountered? If validation is not happening somewhere it's supposed to, yes, we should definitely make a ticket.

Reply:
you are right
I might not have noticed that the message is in the cause